pkg/codegen/dotnet/gen.go

```diff
@@ -1307,7 +1310,9 @@ func (mod *modContext) genResource(w io.Writer, r *schema.Resource) error {
 
 		// Generate result type.
 		if objectReturnType != nil {
-			shouldLiftReturn := mod.liftSingleValueMethodReturns && len(objectReturnType.Properties) == 1
+			objectReturnType, isObjectReturnType := fun.ReturnType.(*schema.ObjectType)
+			shouldLiftReturn := !isObjectReturnType ||
```

Comment:
We check if `objectReturnType != nil`, then redefine `objectReturnType` -- are these the same? If they're different, maybe we could find better names for them?

Reply:
Equally it feels like this could be cleaned up based on the `if` directly above.